packages/mgt-components/src/components/mgt-agenda/mgt-agenda.ts

```diff
@@ -460,8 +459,7 @@ export class MgtAgenda extends MgtTemplatedComponent {
     const grouped = {};
 
     events.forEach(event => {
-      var eventDate = new Date(event.start.dateTime);
-      var dateString = eventDate.toISOString().replace('Z', '');
+      const dateString = event.start.dateTime;
 
       const header = this.getDateHeaderFromDateTimeString(dateString);
```

Comment:
We need to ensure that the dateString passed here is set to UTC.
The dateTimeString returned from the server does not end in Z so needs to have the Z appended when the timeZone is UTC.